### PR TITLE
Add builder pattern for the svm

### PR DESCRIPTION
**Pull request by Aursen**

### CODE REVIEW

**Comment by kevinheavey** [comment]
Now we can't call these methods again after instantiating the LiteSVM struct. For example it's nice to be able to call with_compute_budget() multiple times during tests

**Comment by Aursen** [comment]
> Now we can't call these methods again after instantiating the LiteSVM struct. For example it's nice to be able to call with_compute_budget() multiple times during tests

Let's add a set_compute_budget method, for the rest we don't need it in the svm

**Comment by kevinheavey** [comment]
with_sigverify and with_blockhash_check are also useful to call multiple times

File: src/lib.rs [hunk 1]
```diff
@@ -66,11 +60,14 @@ pub mod error;
 pub mod types;
 
 mod accounts_db;
+mod builder;
 mod builtin;
 mod history;
-mod spl;
+pub mod spl;
```
[the remark this answers is not in the document]
Comment (Aursen) [reply src/lib.rs:66]: For users to be able to load default spl with the builder pattern

[the remark this answers is not in the document]
Comment (kevinheavey) [reply src/lib.rs:66]: But (I think) they don't need access to DEFAULT_SPL_PROGRAMS, they just need a method to add them

File: src/builder.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    }
+
+    /// Sets the checks for the blockhash expiration.
+    pub fn blockhash_check(mut self, check: bool) -> Self {
+        self.svm.blockhash_check = check;
+        self
+    }
+
+    /// Loads programs at the start of the svm.
+    pub fn default_programs(mut self, programs: &[(Pubkey, &[u8])]) -> Self {
```
Comment (kevinheavey) [inline src/builder.rs:68]: this method name is confusing because it actually just does `add_programs`

**Pull request closed without merging**